plexpy/notifiers.py

```diff
@@ -1574,3 +1602,202 @@ def return_config_options(self):
+            for f in files:
+                name, ext = os.path.splitext(f)
+                ext = ext[1:]
+                if ext in ('rb', 'pl', 'bat', 'py', 'sh', 'cmd', 'php'):
+                    fp = os.path.join(scriptdir, f)
+                    scripts[fp] = fp
+
+        return scripts
+
+    def notify(self, subject, message, notify_action=None, script_args=''):
+        logger.debug('Ran notify script subject: %s message: %s, action: %s script_args: %s' % (subject, message, notify_action, script_args))
+
+        prefix = ''
+        script = ''
+
+        if not plexpy.CONFIG.SCRIPTS_FOLDER:
+            return
+
+        # Make sure we use the correct script..
+        if notify_action == 'play':
+            script = plexpy.CONFIG.SCRIPTS_ON_PLAY_SCRIPT
+
+        elif notify_action == 'stop':
+            script = plexpy.CONFIG.SCRIPTS_ON_STOP_SCRIPT
+
+        elif notify_action == 'pause':
+            script = plexpy.CONFIG.SCRIPTS_ON_PAUSE_SCRIPT
+
+        elif notify_action == 'resume':
+            script = plexpy.CONFIG.SCRIPTS_ON_RESUME_SCRIPT
+
+        elif notify_action == 'buffer':
+            script = plexpy.CONFIG.SCRIPTS_ON_BUFFER_SCRIPT
+
+        elif notify_action == 'extdown':
+            script = plexpy.CONFIG.SCRIPTS_ON_EXTDOWN_SCRIPT
+
+        elif notify_action == 'extup':
+            script = plexpy.CONFIG.SCRIPTS_ON_EXTUP_SCRIPT
+
+        elif notify_action == 'intdown':
+            script = plexpy.CONFIG.SCRIPTS_ON_INTDOWN_SCRIPT
+
+        elif notify_action == 'intup':
+            script = plexpy.CONFIG.SCRIPTS_ON_INTUP_SCRIPT
+
+        elif notify_action == 'created':
+            script = plexpy.CONFIG.SCRIPTS_ON_CREATED_SCRIPT
+
+        name, ext = os.path.splitext(script)
+
+        if ext == '.py':
+            prefix = 'python'
+        elif ext == '.php':
+            prefix = 'php'
+        elif ext == '.pl':
+            prefix = 'perl'
+        elif ext == '.rb':
+            prefix = 'ruby'
+
+        script = script.split()
+
+        if prefix:
+            script.insert(0, prefix)
+
+        if script_args:
+            script = script + script_args.split()
+
+        try:
+            p = subprocess.Popen(script, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+                                 stderr=subprocess.STDOUT, cwd=plexpy.CONFIG.SCRIPTS_FOLDER)
+
+            out, error = p.communicate()
+            status = p.returncode
+
+            if out and status:
+                out = out.strip()
+                logger.debug(u'%s returned %s' % (script, out))
+
+            if error:
+                error = error.strip()
+                logger.error(u'%s' % error)
+
+        except OSError as out:
+            logger.error(u'Failed to run %s error %s' % (script, out))
+
+    def return_config_options(self):
+        config_option = [{'label': 'Script folder',
+                          'value': plexpy.CONFIG.SCRIPTS_FOLDER,
```

Comment:
I think you need to trigger `reloadModal()` on the `notifications_config.html` page when the script folder changes to display the script selection drop down values. 

Similar to the Pushbullet API key and the device selection.